### PR TITLE
new package: libmatio

### PR DESCRIPTION
**Pull request by pvonmoradi**
Adds [libmatio](https://github.com/tbeu/matio) for reading and writing binary MATLAB MAT files.

### CODE REVIEW

File: packages/libmatio/build.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+TERMUX_PKG_HOMEPAGE="https://github.com/tbeu/matio"
+TERMUX_PKG_DESCRIPTION="A C library for reading and writing Matlab MAT files"
+TERMUX_PKG_GROUPS="science"
+TERMUX_PKG_LICENSE="BSD 2-Clause"
+TERMUX_PKG_LICENSE_FILE="COPYING"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION="1.5.23"
+TERMUX_PKG_SRCURL="https://github.com/tbeu/matio/archive/refs/tags/v$TERMUX_PKG_VERSION.tar.gz"
+TERMUX_PKG_SHA256=5f25c53e698d06b3119e3a4d0536e1da89983dbb7b0a845270663be02ec95978
+TERMUX_PKG_BUILD_DEPENDS="cmake"
```
Comment (Grimler91) [inline packages/libmatio/build.sh:10]: This line is not needed, cmake is installed by build system automatically. `TERMUX_PKG_BUILD_DEPENDS` means that it downloads cmake that has been cross-compiled for termux, and that cmake cannot be used in any case

File: packages/libmatio/build.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+TERMUX_PKG_HOMEPAGE="https://github.com/tbeu/matio"
+TERMUX_PKG_DESCRIPTION="A C library for reading and writing Matlab MAT files"
+TERMUX_PKG_GROUPS="science"
+TERMUX_PKG_LICENSE="BSD 2-Clause"
+TERMUX_PKG_LICENSE_FILE="COPYING"
```
Comment (Grimler91) [inline packages/libmatio/build.sh:5]: This line shouldn't be needed, COPYING is one of the filenames [termux_step_install_license looks for](https://github.com/termux/termux-packages/blob/master/scripts/build/termux_step_install_license.sh#L35-L42)

**Review by Grimler91 — CHANGES_REQUESTED** [review]
Thanks! Two things to improve

**Comment by pvonmoradi** [comment]
@Grimler91  Why is this building ok in my ubuntu 20.04 instance? 
[log.txt](https://github.com/termux/termux-packages/files/9242632/log.txt)

**Comment by Grimler91** [comment]
Github actions does a "fast build" where it downloads dependencies (`./build-package -i libmatio`) instead of building them, seems downloading pre-built libhdf5 vs building it gives different result.

Seems libmatio looks for static libhdf5 instead of shared library. Probably we want to change so that it used shared library instead with a configure arg or patch, to save disk space for users

**Comment by pvonmoradi** [comment]
@Grimler91 I disabled libhdf5 (and hence support for v7.3 MAT files) with these switches. The build system ignores my directive of using the shared hdf5 libs (.so files) and looks for static (.a) files. 
```cmake
-DMATIO_WITH_HDF5:BOOL=OFF
-DMATIO_MAT73:BOOL=OFF
-DHDF5_USE_STATIC_LIBRARIES:BOOL=OFF
-DMATIO_SHARED:BOOL=ON
```

Here is the set of switches that _should have_ worked but don't:
```cmake
-DMATIO_WITH_HDF5:BOOL=ON
-DMATIO_MAT73:BOOL=ON
-DHDF5_USE_STATIC_LIBRARIES:BOOL=OFF
-DMATIO_SHARED:BOOL=ON
```

File: packages/libmatio/build.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+TERMUX_PKG_GROUPS="science"
+TERMUX_PKG_LICENSE="BSD 2-Clause"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION="1.5.23"
+TERMUX_PKG_SRCURL="https://github.com/tbeu/matio/archive/refs/tags/v$TERMUX_PKG_VERSION.tar.gz"
+TERMUX_PKG_SHA256=5f25c53e698d06b3119e3a4d0536e1da89983dbb7b0a845270663be02ec95978
+TERMUX_PKG_DEPENDS="zlib"
+TERMUX_PKG_BUILD_IN_SRC=true
+TERMUX_PKG_AUTO_UPDATE=true
+TERMUX_PKG_EXTRA_CONFIGURE_ARGS='
```
Comment (Grimler91) [inline packages/libmatio/build.sh:12]: Please change to " here, no need for single quotes

**Review by Grimler91 — COMMENTED** [review]
One last cosmetic change, then I'll merge it!

**Pull request merged**